geom/wkb_test.go

```diff
@@ -678,3 +679,46 @@ func TestWKBUnmarshalEndianess(t *testing.T) {
 		})
 	}
 }
+
+func TestWKBGeometryCollectionMixedCoordinateTypes(t *testing.T) {
+	const (
+		bigEndian = "00"
+		f64Zero   = "0000000000000000"
+		u32One    = "00000001"
+	)
+
+	for _, gc := range []struct {
+		hex string
+		ct  geom.CoordinatesType
+	}{
+		{"00000007", geom.DimXY},
+		{"000003ef", geom.DimXYZ},
+		{"000007d7", geom.DimXYM},
+		{"00000bbf", geom.DimXYZM},
+	} {
+		for _, point := range []struct {
+			hex string
+			ct  geom.CoordinatesType
+		}{
+			{"00000001", geom.DimXY},
+			{"000003e9", geom.DimXYZ},
+			{"000007d1", geom.DimXYM},
+			{"00000bb9", geom.DimXYZM},
+		} {
+			t.Run(fmt.Sprintf("gc_%v_point_%v", gc.ct, point.ct), func(t *testing.T) {
+				hex := bigEndian +
+					gc.hex +
+					u32One +
+					bigEndian +
+					point.hex +
+					strings.Repeat(f64Zero, point.ct.Dimension())
+				_, err := geom.UnmarshalWKB(hexStringToBytes(t, hex))
+				if gc.ct == point.ct {
+					expectNoErr(t, err)
+				} else {
+					expectErr(t, err)
```

Comment:
Should we assert that the error type is `mismatchedGeometryCollectionDimsError`?

Reply:
Actually.... maybe there is a way to get the best of both worlds. I'll do some experimentation and raise a fix in a new PR if I can get it working.

Reply:
I tried something out in https://github.com/peterstace/simplefeatures/pull/616 (it seems to work pretty well, would love to get your opinion.